c10/util/LeftRight.h

```diff
@@ -204,14 +203,17 @@ class RWSafeLeftRightWrapper final {
   RWSafeLeftRightWrapper& operator=(RWSafeLeftRightWrapper&&) noexcept = delete;
 
   template <typename F>
-  auto read(F&& readFunc) const -> typename c10::invoke_result_t<F, const T&> {
+  // NOLINTNEXTLINE(cppcoreguidelines-missing-std-forward)
+  auto read(F&& readFunc) const {
     return data_.withLock(
-        [&readFunc](T const& data) { return readFunc(data); });
+        [&readFunc](T const& data) { return std::forward<F>(readFunc)(data); });
   }
 
   template <typename F>
-  auto write(F&& writeFunc) -> typename c10::invoke_result_t<F, T&> {
-    return data_.withLock([&writeFunc](T& data) { return writeFunc(data); });
+  // NOLINTNEXTLINE(cppcoreguidelines-missing-std-forward)
+  auto write(F&& writeFunc) {
+    return data_.withLock(
+        [&writeFunc](T& data) { return std::forward<F>(writeFunc)(data); });
```

Comment:
Wait, this isn't quite right... you are forwarding a reference, not a universal reference. If you want to do this right, it needs to be a move capture in the `[]. I think nearly all the std::forward fixes in this file seems like false positives though.

Reply:
Lambda capture and perfect forward is somewhat confusing. But I think this fix is correct as you can see from https://stackoverflow.com/questions/21238463/capture-by-universal-reference

Reply:
@Skylion007 I think the fix is correct. The function was immediately called with lock, so there is no danger of dangling reference. And https://stackoverflow.com/questions/21238463/capture-by-universal-reference explains well the syntax here.

Reply:
I am more worried about `[&writeFunc]` the writeFunc in the lambda will always be an lvalue right? so the std::forward is safe, but a noop.